From af3ceff3108a8470b35e3bc78b9d31ff4bd05e91 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 30 Sep 2014 10:50:27 +0100
Subject: [PATCH] RHEL: v2v: Disable the --qemu-boot / -oo qemu-boot option
 (RHBZ#1147313).

This cannot work because there is no Gtk or SDL output mode
in RHEL's qemu-kvm.

In addition you will have to edit the -display option in the
qemu script.
---
 docs/virt-v2v-output-local.pod |  6 ++----
 docs/virt-v2v.pod              | 13 -------------
 output/output_qemu.ml          |  3 +++
 3 files changed, 5 insertions(+), 17 deletions(-)

diff --git a/docs/virt-v2v-output-local.pod b/docs/virt-v2v-output-local.pod
index 49f00754..bdf12c5d 100644
--- a/docs/virt-v2v-output-local.pod
+++ b/docs/virt-v2v-output-local.pod
@@ -9,7 +9,7 @@ or libvirt
 
  virt-v2v [-i* options] -o local -os DIRECTORY
 
- virt-v2v [-i* options] -o qemu -os DIRECTORY [-oo qemu-boot]
+ virt-v2v [-i* options] -o qemu -os DIRECTORY
 
  virt-v2v [-i* options] -o null
 
@@ -47,12 +47,10 @@ where C<NAME> is the guest name.
 
 =item B<-o qemu -os> C<DIRECTORY>
 
-=item B<-o qemu -os> C<DIRECTORY> B<-oo qemu-boot>
-
 This converts the guest to files in C<DIRECTORY>.  Unlike I<-o local>
 above, a shell script is created which contains the raw qemu command
 you would need to boot the guest.  However the shell script is not
-run, I<unless> you also add the I<-oo qemu-boot> option.
+run.
 
 =item B<-o null>
 
diff --git a/docs/virt-v2v.pod b/docs/virt-v2v.pod
index ecb29771..41bb58ce 100644
--- a/docs/virt-v2v.pod
+++ b/docs/virt-v2v.pod
@@ -169,11 +169,6 @@ Since F<guest-domain.xml> contains the path(s) to the guest disk
 image(s) you do not need to specify the name of the disk image on the
 command line.
 
-To convert a local disk image and immediately boot it in local
-qemu, do:
-
- virt-v2v -i disk disk.img -o qemu -os /var/tmp -oo qemu-boot
-
 =head1 OPTIONS
 
 =over 4
@@ -585,9 +580,6 @@ This is similar to I<-o local>, except that a shell script is written
 which you can use to boot the guest in qemu.  The converted disks and
 shell script are written to the directory specified by I<-os>.
 
-When using this output mode, you can also specify the I<-oo qemu-boot>
-option which boots the guest under qemu immediately.
-
 ifelse(ENABLE_OVIRT, yes, `
 =item B<-o> B<vdsm>
 
@@ -660,11 +652,6 @@ For I<-o openstack> (L<virt-v2v-output-openstack(1)>) only, set a guest ID
 which is saved on each Cinder volume in the C<virt_v2v_guest_id>
 volume property.
 
-=item B<-oo qemu-boot>
-
-When using I<-o qemu> only, this boots the guest immediately after
-virt-v2v finishes.
-
 =item B<-oo os->*B<=>*
 
 For I<-o openstack> (L<virt-v2v-output-openstack(1)>) only, set optional
diff --git a/output/output_qemu.ml b/output/output_qemu.ml
index b17ac518..197fe385 100644
--- a/output/output_qemu.ml
+++ b/output/output_qemu.ml
@@ -65,6 +65,9 @@ module QEMU = struct
     let compressed = !compressed
     and qemu_boot = !qemu_boot in
 
+    if qemu_boot then
+      error (f_"-o qemu: the -oo qemu-boot option cannot be used in RHEL");
+
     (* -os must be set to a directory. *)
     let output_storage =
       match options.output_storage with
